src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -69,38 +76,38 @@ void EnSw_CrossProduct(Vec3f* a, Vec3f* b, Vec3f* dst) {
     dst->z = (a->x * b->y) - (a->y * b->x);
 }
 
-s32 func_80B0BE20(EnSw* this, CollisionPoly* poly) {
+s32 EnSw_ClingToWall(EnSw* this, CollisionPoly* poly) {
     Vec3f polyNormal;
     Vec3f sp38;
-    f32 sp34;
-    f32 temp_f0;
+    f32 dot;
+    f32 length;
     s32 pad;
 
     this->actor.floorPoly = poly;
     polyNormal.x = COLPOLY_GET_NORMAL(poly->normal.x);
     polyNormal.y = COLPOLY_GET_NORMAL(poly->normal.y);
     polyNormal.z = COLPOLY_GET_NORMAL(poly->normal.z);
-    sp34 = Math_FAcosF(DOTXYZ(polyNormal, this->unk_364));
-    EnSw_CrossProduct(&this->unk_364, &polyNormal, &sp38);
-    Matrix_RotateAxis(sp34, &sp38, MTXMODE_NEW);
+    dot = Math_FAcosF(DOTXYZ(polyNormal, this->normalVec));
+    EnSw_CrossProduct(&this->normalVec, &polyNormal, &sp38);
+    Matrix_RotateAxis(dot, &sp38, MTXMODE_NEW);
     Matrix_MultVec3f(&this->unk_370, &sp38);
     this->unk_370 = sp38;
     EnSw_CrossProduct(&this->unk_370, &polyNormal, &this->unk_37C);
-    temp_f0 = Math3D_Vec3fMagnitude(&this->unk_37C);
-    if (temp_f0 < 0.001f) {
+    length = Math3D_Vec3fMagnitude(&this->unk_37C);
+    if (length < 0.001f) {
         return 0;
     }
-    this->unk_37C.x = this->unk_37C.x * (1.0f / temp_f0);
-    this->unk_37C.y = this->unk_37C.y * (1.0f / temp_f0);
-    this->unk_37C.z = this->unk_37C.z * (1.0f / temp_f0);
-    this->unk_364 = polyNormal;
+    this->unk_37C.x = this->unk_37C.x * (1.0f / length);
+    this->unk_37C.y = this->unk_37C.y * (1.0f / length);
+    this->unk_37C.z = this->unk_37C.z * (1.0f / length);
+    this->normalVec = polyNormal;
```

Comment:
I think `normalVec` -> `wallNolyNormal` would be more descriptive (if that's what it is)

Reply:
just making sure, typo for wallPoly right?

Reply:
Yes, whoops